### ISSUE TITLE
[Bug] different hidden_states with different precompile-token-paddings

### ISSUE THREAD

**Issue by pathfinder-pf** [sgl-project/sglang-jax#334]
### Checklist

- [x] 1. I have searched related issues but cannot get the expected help.
- [x] 2. The bug has not been fixed in the latest version.
- [x] 3. Please note that if the bug-related issue you submitted lacks corresponding environment info and a minimal reproducible demo, it will be challenging for us to reproduce and resolve the issue, reducing the likelihood of receiving feedback.
- [x] 4. If the issue you raised is not a bug but a question, please raise a discussion at https://github.com/sgl-project/sgl-jax/discussions/new/choose Otherwise, it will be closed.
- [x] 5. Please use English, otherwise it will be closed.

### Describe the bug

when use different precompile-token-paddings，the hidden states is different

### Reproduction
commit bce2119fe54400726578caa6913e2df7f4adb7a3
```
pip install jax==0.7.1 flax==0.12.0 libtpu==0.0.17
```
- launch server
JAX_COMPILATION_CACHE_DIR=/tmp/jit_cache python3 -u -m sgl_jax.launch_server \
--model-path deepseek-ai/DeepSeek-R1-Distill-Qwen-1.5B \
--trust-remote-code  \
--tp-size=1 \
--device=tpu \
--mem-fraction-static=0.8 \
--chunked-prefill-size=2048 \
--download-dir=/tmp \
--dtype=bfloat16 \
--max-running-requests 128 \
--skip-server-warmup \
--page-size=128 \
--use-sort-for-toppk-minp \
--enable-precision-tracer \
--precompile-bs-paddings=64 \
--max-total-tokens=257536 \
--precompile-token-paddings=8192 \  ## remove this parameter, you will see the difference
--disable-radix-cache \
--disable-overlap-schedule

- start trace
curl -H "Content-Type: application/json" -X POST  http://localhost:30000/start_trace -d '{"req_num":1}'
- eval
evalscope eval  --model deepseek-ai/DeepSeek-R1-Distill-Qwen-1.5B --api-url http://127.0.0.1:30000/v1/chat/completions --api-key EMPTY --eval-type service --datasets aime24  --eval-batch-size 64 --dataset-args '{"aime24":{"metric_list":["Pass@1"]}}' --generation-config '{"temperature": 0.6, "top_k":1, "n":1, "max_tokens": 1}' --timeout 120000 --limit 1

- compare two result file
```
import json
with open('./goo.jsonl') as f:
    bad = json.loads(f.read())

with open('./bad.jsonl') as f:
    goo = json.loads(f.read())

bad_prefill_records = sorted(bad["precision_records"]["prefill"][0]["records"], key=lambda x: (x["layer_id"], x["stage"], x["name"]))
goo_prefill_records = sorted(goo["precision_records"]["prefill"][0]["records"], key=lambda x: (x["layer_id"], x["stage"], x["name"]))
for p, (b, g) in enumerate(zip(bad_prefill_records, goo_prefill_records)):
    if b["value"] != g["value"]:
        del b["value"]
        del g["value"]
        print(p, b, g)
```
### Environment

v6e-1
